volmdlr/faces.py

```diff
@@ -4817,6 +4816,9 @@ def __init__(self, surface3d, surface2d: Surface2D,
                  name: str = ''):
         self.surface3d = surface3d
         self.surface2d = surface2d
+        self._outer_contour3d = None
+        self._inner_contours3d = None
+        # self.bounding_box = self._bounding_box()
```

Comment:
Created Cache variables `self._outer_contour3d` and `self._inner_contours3d` so we can calculate these parameters once and use them whenever we need. The main objective is to have a gain in performance. they're being used in the corresponding property methods.